src/app/pages/PerpetualPage/hooks/usePerpetual_AmmDepthChart.ts

```diff
@@ -29,20 +24,27 @@ export type AmmDepthChartData = {
 export const usePerpetual_AmmDepthChart = (
   pair: PerpetualPair,
 ): AmmDepthChartData => {
-  const { ammState, perpetualParameters } = useContext(PerpetualQueriesContext);
+  const { ammState, depthMatrixEntries: entries } = useContext(
+    PerpetualQueriesContext,
+  );
   const { trades } = useContext(RecentTradesContext);
 
   const data = useMemo(() => {
     const indexPrice = getIndexPrice(ammState);
     const markPrice = getMarkPrice(ammState);
-    const entries = getDepthMatrix(perpetualParameters, ammState);
+
+    let averagePrice = undefined;
 
     let shorts: AmmDepthChartDataEntry[] = [];
     let longs: AmmDepthChartDataEntry[] = [];
 
     if (entries && entries.length >= 3) {
       const length = entries[0].length;
       const midIndex = Math.floor(length / 2);
+
+      const averagePriceIndex = entries[1].indexOf(0);
+      averagePrice = entries[0][averagePriceIndex];
```

Comment:
your averagePriceIndex should be the same as midIndex.
either use midIndex or your indexing approach. not both.
```suggestion
      averagePrice = entries[0][midIndex];
```